File: data-source/sort/src/com/SelectSort.java
package com;

import java.util.Arrays;

public class SelectSort {
    public static int[] selectsort(int[] array){
        if(array==null){
            return null;
        }
        int[] arr = Arrays.copyOf(array,array.length);
        int minIndex = 0;
        for(int i=0; i<arr.length-1; i++){//n-1趟，每趟选出一个最xiao的数，交换一次
            minIndex = i;
            for(int j=i+1; j<arr.length; j++){
                if(arr[j]<arr[minIndex]){
                    minIndex=j;
                }
            }
            //swap
            if(minIndex!=i){
                int t = arr[i];
                arr[i] = arr[minIndex];
                arr[minIndex] =t;
            }

        }
        return arr;
    }
}

File: sort/selectSort.cpp
void swap(vector<int> &arr, int i, int j){
    int tmp = arr[i];
    arr[i] = arr[j];
    arr[j] = tmp;
}
selectSort(vector<int> &arr){
    if(arr.empty() || arr.size()<2) return;
    for(int i=0; i<arr.size()-1; i++){
        int minIndex = i;
	for(int j=i+1; j<arr.size(); j++){
	    if(arr[j]<arr[minIndex]){
	        minIndex = j;
	    }
	}
	swap(arr, i, minIndex);
    }
    return;
}
